plugins/schedule/base/lib/api/KalturaScheduleEvent.php

```diff
@@ -292,13 +307,33 @@ public function validateForInsert($propertiesToSkip = array())
 	{
 		$this->validatePropertyNotNull('recurrenceType');
 		$this->validatePropertyNotNull('summary');
-		$this->validatePropertyNotNull('startDate');
+		if (is_null($this->startDate) && is_null($this->linkedTo))
+		{
+			throw new KalturaAPIException(KalturaScheduleErrors::START_TIME_AND_LINKED_TO_NOT_SET);
+		}
+		if (!is_null($this->startDate) && !is_null($this->linkedTo))
+		{
+			throw new KalturaAPIException(KalturaScheduleErrors::START_TIME_AND_LINKED_TO_CONFLICT);
+		}
+		if (!is_null($this->linkedTo) && $this->recurrenceType != KalturaScheduleEventRecurrenceType::NONE)
+		{
+			throw new KalturaAPIException(KalturaScheduleErrors::RECURRENCE_LINKED_EVENT_CONFLICT);
+		}
+		if (!is_null($this->startDate))
+		{
+			$this->validatePropertyNotNull('endDate');
+		}
+		if (!is_null($this->linkedTo))
+		{
+			// retrieve relevant event's start time and calculate the current event start and end time
+			$this->validatePropertyNotNull('duration');
+			$this->setTimingFromLinkedToEvent($this->linkedTo->getEventId());

```

Comment:
Since we have duration here, pass it to  setTimingFromLinkedToEvent so we would need to take duration from the object